src/reformatters/common/shared_memory_utils.py

```diff
@@ -96,9 +98,9 @@ def create_data_array_and_template(
         buffer=shared_buffer.buf,
     )
     # Important:
-    # We rely on initializing with nans so failed reads (eg. corrupt source data)
-    # leave nan and to reuse the same shared buffer for each variable.
-    shared_array[:] = np.nan
+    # We rely on initializing with a fill value so failed reads (eg. corrupt source data)
+    # leave the fill value and to reuse the same shared buffer for each variable.
+    shared_array[:] = fill_value
```

Comment:
Other half the change i mentioned in the other comment to handle int16s which can't have nans.